.github/workflows/3rd_party_lint.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-FileCopyrightText: 2023 DB Systel GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# Build reuse-tool and lint 3rd party repositories for which we know that they
+# are reliably REUSE compliant, rather complex, use several annotation
+# strategies, and are quite popular. This shall prevent that we introduce
+# unforeseen and unintended breaking changes.
+
+name: Lint 3rd party repositories
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  third-party-lint:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.ignore-failure }}
+    strategy:
+      # do not abort the whole test job if one step fails
+      fail-fast: false
+      #
+      matrix:
+        repo: ["fsfe/reuse-example", "curl/curl", "SAP/openui5"]
+        ignore-failure: [false]
+        # add a repo for which we know it's not compliant
+        include:
+          - repo: spdx/license-list-XML
+            ignore-failure: true
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry install --no-interaction --no-dev
+      # Clone and lint repositories
+      - name: Clone ${{ matrix.repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ matrix.repo }}
+          path: 3rd/${{ matrix.repo }}
+      - name: Lint ${{ matrix.repo }}
+        run: poetry run reuse --root 3rd/${{ matrix.repo }} lint
```

Comment:
I think it'd be nice if we instead called a script that does this, so that we can call the script locally during testing.

Reply:
Makes sense. But I wonder where to clone the repos to. Shall we establish the folder "3rd" as an ignored one that can be used for this?

Reply:
Maybe using https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory (uses OS default equivalent to `/tmp/[random-hash-or-whatever]/`, but the downside is that typically the directory is deleted when the process ends, which means:

- Can't investigate the directory after the test failed, because the directory is deleted.
- More clones than needed if running successively (but using existing git clones would imply more complex logic; cloning every time is a lot easier).

So maybe `[tests/]3rd/` like you propose.

Alternatively: `~/.cache/reuse/[...]/` (using `appdirs` dependency to find OS-agnostic XDG_CACHE_HOME equivalent).

Reply:
So I went for the very naive path of just creating a relative directory for these things. A temporary dir would have been more elegant but I figured that this may not work in the Github runner (but I did not try TBH).